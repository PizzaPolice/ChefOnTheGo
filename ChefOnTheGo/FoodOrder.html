<!DOCTYPE html>
<html>
    <head>
        <meta charset="utf-8" />


        <meta name="viewport" content="width=device-width, initial-scale=1">

        <title>Shopping List</title>

        <link rel="stylesheet" href="FoodOrderStyle.css" media="screen" title="no title" charset="utf-8">

        <script src="https://code.jquery.com/jquery-2.2.4.js" charset="utf-8"></script>

        <meta name="robots" content="noindex,follow">

    </head>
 

<body>
    <div class="shopping-cart">
        <!-- Title -->
        <div class="title">
            Food Order
        </div>

        <!-- Product #1 -->
        <div class="item">
            <div class="buttons">
                <span class="delete-btn"></span>
                <span class="like-btn"></span>
            </div>

            <div class="image">
                <img src="tortilla soup.png" alt="" />
            </div>

            <div class="description">
                <span>Tortilla Soup</span>
               
            </div>

            <div class="quantity">
                <button class="plus-btn" type="button" name="button">
                    <img src="plus.svg" alt="" />
                </button>
                <input type="text" name="name" value="1">
                <button class="minus-btn" type="button" name="button">
                    <img src="minus.svg" alt="" />
                </button>
            </div>

            <div class="total-price">$8.50</div>
        </div>

        <!-- Product #2 -->
        <div class="item">
            <div class="buttons">
                <span class="delete-btn"></span>
                <span class="like-btn"></span>
            </div>

            <div class="image">
                <img src="pineapple fajitas.png" alt="" />
            </div>

            <div class="description">
                <span>Pineapple Fajitas</span>
            </div>

            <div class="quantity">
                <button class="plus-btn" type="button" name="button">
                    <img src="plus.svg" alt="" />
                </button>
                <input type="text" name="name" value="1">
                <button class="minus-btn" type="button" name="button">
                    <img src="minus.svg" alt="" />
                </button>
            </div>

            <div class="total-price">$13.50</div>
        </div>

        <!-- Product #3 -->
        <div class="item">
            <div class="buttons">
                <span class="delete-btn"></span>
                <span class="like-btn"></span>
            </div>

            <div class="image">
                <img src="guac.png" alt="" />
            </div>

            <div class="description">
                <span>Guacamole</span>
            </div>

            <div class="quantity">
                <button class="plus-btn" type="button" name="button">
                    <img src="plus.svg" alt="" />
                </button>
                <input type="text" name="name" value="1">
                <button class="minus-btn" type="button" name="button">
                    <img src="minus.svg" alt="" />
                </button>
            </div>

            <div class="total-price">$4.50</div>
        </div>
    </div>




</body>
</html>