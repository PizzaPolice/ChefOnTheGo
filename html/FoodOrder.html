<!DOCTYPE html>


 <!-- Top Navigation --> 
<html>
<head>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style>
    body {margin:0;}

    .topnav {
      overflow: hidden;
      background-color: #333;
    }

    .topnav a {
      float: left;
      display: block;
      color: #f2f2f2;
      text-align: center;
      padding: 14px 16px;
      text-decoration: none;
      font-size: 17px;
    }

    .topnav a:hover {
      background-color: #ddd;
      color: black;
    }

    .active {
      background-color: #21618C;
      color: white;
    }

    .topnav .icon {
      display: none;
    }

    @media screen and (max-width: 600px) {
      .topnav a:not(:first-child) {display: none;}
      .topnav a.icon {
        float: right;
        display: block;
      }
    }

    @media screen and (max-width: 600px) {
      .topnav.responsive {position: relative;}
      .topnav.responsive .icon {
        position: absolute;
        right: 0;
        top: 0;
      }
      .topnav.responsive a {
        float: none;
        display: block;
        text-align: left;
      }

    }
    </style>
</head>



<body>

    <div class="topnav" id="myTopnav">
        <a href="#home" class="active">Home</a>
        <a href="FoodOrder.html">Food Order</a>
        <a href="Checkout.html">Checkout</a>
        <a href="OrderStatus.html">Order Status</a>
        <a href="javascript:void(0);" style="font-size:15px;" class="icon" onclick="myFunction()">&#9776;</a>
    </div>

    <div style="padding-left:16px">

    </div>

    <script>
        function myFunction() {
            var x = document.getElementById("myTopnav");
            if (x.className === "topnav") {
                x.className += " responsive";
            } else {
                x.className = "topnav";
            }
        }
    </script>

</body>
</html>




 <!-- Food Order --> 

<html>
    <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>Shopping List</title>
        <link rel="stylesheet" href="FoodOrderStyle.css" media="screen" title="no title" charset="utf-8">
        <script src="https://code.jquery.com/jquery-2.2.4.js" charset="utf-8"></script>
        <meta name="robots" content="noindex,follow">

    </head>
 

<body>
    <div class="shopping-cart">
        <!-- Title -->
        <div class="title">
            Food Order
        </div>

        <!-- Product #1 -->
        <div class="item">
            <div class="buttons">
                <span class="delete-btn"></span>
                <span class="like-btn"></span>
            </div>

            <div class="image">
                <img src="tortilla soup.png" alt="" />
            </div>

            <div class="description">
                <span>Tortilla Soup</span>
               
            </div>

            <div class="quantity">
                <button class="plus-btn" type="button" name="button">
                    <img src="plus.svg" alt="" />
                </button>
                <input type="text" name="name" value="1">
                <button class="minus-btn" type="button" name="button">
                    <img src="minus.svg" alt="" />
                </button>
            </div>

            <div class="total-price">$8.50</div>
        </div>

        <!-- Product #2 -->
        <div class="item">
            <div class="buttons">
                <span class="delete-btn"></span>
                <span class="like-btn"></span>
            </div>

            <div class="image">
                <img src="pineapple fajitas.png" alt="" />
            </div>

            <div class="description">
                <span>Pineapple Fajitas</span>
            </div>

            <div class="quantity">
                <button class="plus-btn" type="button" name="button">
                    <img src="plus.svg" alt="" />
                </button>
                <input type="text" name="name" value="1">
                <button class="minus-btn" type="button" name="button">
                    <img src="minus.svg" alt="" />
                </button>
            </div>

            <div class="total-price">$13.50</div>
        </div>

        <!-- Product #3 -->
        <div class="item">
            <div class="buttons">
                <span class="delete-btn"></span>
                <span class="like-btn"></span>
            </div>

            <div class="image">
                <img src="guac.png" alt="" />
            </div>

            <div class="description">
                <span>Guacamole</span>
            </div>

            <div class="quantity">
                <button class="plus-btn" type="button" name="button">
                    <img src="plus.svg" alt="" />
                </button>
                <input type="text" name="name" value="1">
                <button class="minus-btn" type="button" name="button">
                    <img src="minus.svg" alt="" />
                </button>
            </div>

            <div class="total-price">$4.50</div>
        </div>
    </div>



 <!-- Javascript --> 

    <script type="text/javascript">

      $('.minus-btn').on('click', function(e) {
            e.preventDefault();
            var $this = $(this);
            var $input = $this.closest('div').find('input');
            var value = parseInt($input.val());

            if (value > 1) {
                value = value - 1;
            } else {
                value = 0;
            }

        $input.val(value);

        });

        $('.plus-btn').on('click', function(e) {
            e.preventDefault();
            var $this = $(this);
            var $input = $this.closest('div').find('input');
            var value = parseInt($input.val());

            if (value < 100) {
            value = value + 1;
            } else {
                value =100;
            }

            $input.val(value);
        });

      $('.like-btn').on('click', function() {
        $(this).toggleClass('is-active');
      });


        </script>

</body>
</html>