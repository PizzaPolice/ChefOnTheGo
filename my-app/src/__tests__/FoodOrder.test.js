import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import FoodOrder from '../components/food_order.jsx'
import { createStore } from 'redux'

//Mock store
store = createStore();

it('EditDishes component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={store}><FoodOrder /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
  });


describe('Login Component', () => {
    it('Login should render without throwing an error', () => {
        expect(shallow(<FoodOrder />).exists(<form className='Login'></form>)).toBe(true)
      })
      it("should render initial layout", () => {
        const component = shallow(<FoodOrder />);
        expect(component.getElements()).toMatchSnapshot();
        })
    });

    describe('Submit Button', () => {
        it('should be defined', () => {
          expect(FoodOrder).toBeDefined();
        });

        it('should render correctly', () => {
          const button = shallow(
            <FoodOrder name='btn next-btn' />
          );
          expect(button).toMatchSnapshot();
        });
    });
