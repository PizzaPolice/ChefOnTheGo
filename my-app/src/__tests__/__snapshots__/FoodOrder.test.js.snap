// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should render initial layout 1`] = `
Array [
  <div
    className="outer-panel"
>
    <div
        className="panel-body"
    >
        <div
            className="title"
        >
            Food Order
        </div>
        <FoodList
            foods={
                Array [
                    Object {
                      "name": "food1",
                      "price": "100",
                    },
                    Object {
                      "name": "food2",
                      "price": "101",
                    },
                  ]
            }
        />
        <div
            className="panel-footer"
        >
            <Link
                replace={false}
                to="list_chefs"
            >
                <button
                    className="btn back-btn"
                >
                    Back
                </button>
            </Link>
            <Link
                onClick={[Function]}
                replace={false}
                to="checkout"
            >
                <button
                    className="btn next-btn"
                >
                    Submit Order
                </button>
            </Link>
        </div>
    </div>
</div>,
]
`;

exports[`Submit Button should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FoodOrder
    name="btn next-btn"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="panel-body"
>
        <div
                className="title"
        >
                Food Order
        </div>
        <FoodList
                foods={
                        Array [
                                Object {
                                  "name": "food1",
                                  "price": "100",
                                },
                                Object {
                                  "name": "food2",
                                  "price": "101",
                                },
                              ]
                }
        />
        <div
                className="panel-footer"
        >
                <Link
                        replace={false}
                        to="list_chefs"
                >
                        <button
                                className="btn back-btn"
                        >
                                Back
                        </button>
                </Link>
                <Link
                        onClick={[Function]}
                        replace={false}
                        to="checkout"
                >
                        <button
                                className="btn next-btn"
                        >
                                Submit Order
                        </button>
                </Link>
        </div>
</div>,
      "className": "outer-panel",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
>
            Food Order
</div>,
          <FoodList
            foods={
                        Array [
                                    Object {
                                      "name": "food1",
                                      "price": "100",
                                    },
                                    Object {
                                      "name": "food2",
                                      "price": "101",
                                    },
                                  ]
            }
/>,
          <div
            className="panel-footer"
>
            <Link
                        replace={false}
                        to="list_chefs"
            >
                        <button
                                    className="btn back-btn"
                        >
                                    Back
                        </button>
            </Link>
            <Link
                        onClick={[Function]}
                        replace={false}
                        to="checkout"
            >
                        <button
                                    className="btn next-btn"
                        >
                                    Submit Order
                        </button>
            </Link>
</div>,
        ],
        "className": "panel-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Food Order",
            "className": "title",
          },
          "ref": null,
          "rendered": "Food Order",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "foods": Array [
              Object {
                "name": "food1",
                "price": "100",
              },
              Object {
                "name": "food2",
                "price": "101",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                replace={false}
                to="list_chefs"
>
                <button
                                className="btn back-btn"
                >
                                Back
                </button>
</Link>,
              <Link
                onClick={[Function]}
                replace={false}
                to="checkout"
>
                <button
                                className="btn next-btn"
                >
                                Submit Order
                </button>
</Link>,
            ],
            "className": "panel-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn back-btn"
>
                  Back
</button>,
                "replace": false,
                "to": "list_chefs",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Back",
                  "className": "btn back-btn",
                },
                "ref": null,
                "rendered": "Back",
                "type": "button",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn next-btn"
>
                  Submit Order
</button>,
                "onClick": [Function],
                "replace": false,
                "to": "checkout",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Submit Order",
                  "className": "btn next-btn",
                },
                "ref": null,
                "rendered": "Submit Order",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="panel-body"
>
          <div
                    className="title"
          >
                    Food Order
          </div>
          <FoodList
                    foods={
                              Array [
                                        Object {
                                          "name": "food1",
                                          "price": "100",
                                        },
                                        Object {
                                          "name": "food2",
                                          "price": "101",
                                        },
                                      ]
                    }
          />
          <div
                    className="panel-footer"
          >
                    <Link
                              replace={false}
                              to="list_chefs"
                    >
                              <button
                                        className="btn back-btn"
                              >
                                        Back
                              </button>
                    </Link>
                    <Link
                              onClick={[Function]}
                              replace={false}
                              to="checkout"
                    >
                              <button
                                        className="btn next-btn"
                              >
                                        Submit Order
                              </button>
                    </Link>
          </div>
</div>,
        "className": "outer-panel",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
>
              Food Order
</div>,
            <FoodList
              foods={
                            Array [
                                          Object {
                                            "name": "food1",
                                            "price": "100",
                                          },
                                          Object {
                                            "name": "food2",
                                            "price": "101",
                                          },
                                        ]
              }
/>,
            <div
              className="panel-footer"
>
              <Link
                            replace={false}
                            to="list_chefs"
              >
                            <button
                                          className="btn back-btn"
                            >
                                          Back
                            </button>
              </Link>
              <Link
                            onClick={[Function]}
                            replace={false}
                            to="checkout"
              >
                            <button
                                          className="btn next-btn"
                            >
                                          Submit Order
                            </button>
              </Link>
</div>,
          ],
          "className": "panel-body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Food Order",
              "className": "title",
            },
            "ref": null,
            "rendered": "Food Order",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "foods": Array [
                Object {
                  "name": "food1",
                  "price": "100",
                },
                Object {
                  "name": "food2",
                  "price": "101",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Link
                  replace={false}
                  to="list_chefs"
>
                  <button
                                    className="btn back-btn"
                  >
                                    Back
                  </button>
</Link>,
                <Link
                  onClick={[Function]}
                  replace={false}
                  to="checkout"
>
                  <button
                                    className="btn next-btn"
                  >
                                    Submit Order
                  </button>
</Link>,
              ],
              "className": "panel-footer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="btn back-btn"
>
                    Back
</button>,
                  "replace": false,
                  "to": "list_chefs",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Back",
                    "className": "btn back-btn",
                  },
                  "ref": null,
                  "rendered": "Back",
                  "type": "button",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <button
                    className="btn next-btn"
>
                    Submit Order
</button>,
                  "onClick": [Function],
                  "replace": false,
                  "to": "checkout",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Submit Order",
                    "className": "btn next-btn",
                  },
                  "ref": null,
                  "rendered": "Submit Order",
                  "type": "button",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
